{
 "metadata": {
  "kernelspec": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "display_name": "IPython (Python 2)",
   "language": "python",
   "name": "python2"
  },
  "language": "Julia",
  "name": "",
  "signature": "sha256:6cba4f36274d6d118711a6b6e13862d829ac9dd94bdde704a3103a03fcb1ae48"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Single-Step GBLUP\n",
      "=================\n",
      "\n",
      "Introduction\n",
      "------------\n",
      "\n",
      "-   Genotypes are available only on a few thousand (non-random)\n",
      "    individuals\n",
      "\n",
      "-   Phenotype and pedigree information available on millions\n",
      "\n",
      "-   Often, phenotypes are not available on genotyped individuals (sires)\n",
      "\n",
      "-   Training (estimation of marker effects) based on deregressed EBV\n",
      "\n",
      "-   Marker-based EBV combined with pedigree-based EBV using selection\n",
      "    index theory\n",
      "\n",
      "-   An alternative, single-step approach proposed by Legarra et al.\n",
      "    @Legarra:2009:J-Dairy-Sci:19700729\n",
      "    [@Misztal:2009:J-Dairy-Sci:19700728; @Aguilar:2010:J-Dairy-Sci:20105546]"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Theory\n",
      "------\n",
      "\n",
      "### Marker Effect Model\n",
      "\n",
      "$$\\mathbf{y}=\\mathbf{X}\\boldsymbol{\\beta}+\\mathbf{M}\\boldsymbol{\\alpha}+\\mathbf{e},$$\n",
      "\n",
      "-   $\\boldsymbol{\\beta}$ are fixed effects\n",
      "\n",
      "-   $\\mathbf{X}$ incidence matrix for fixed effects\n",
      "\n",
      "-   $\\mathbf{M}$ marker covariates\n",
      "\n",
      "-   $\\boldsymbol{\\alpha}\\sim N(\\mathbf{0},\\mathbf{I}\\sigma_{\\alpha}^{2})$\n",
      "\n",
      "-   $\\mathbf{e}\\sim N(\\mathbf{0},\\mathbf{I\\sigma_{e}^{2})}$"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "### Breeding Value Models\n",
      "\n",
      "-   Two mixed linear models are linearly equivalent and will lead to the\n",
      "    same inferences if the vector $\\mathbf{y}$ of observations has the\n",
      "    same first and second moments in both models @Henderson.CR:1984.\n",
      "\n",
      "-   In this sense, a model that is equivalent to ([eq:2-1]) can be\n",
      "    written as\n",
      "\n",
      "    $$\\mathbf{y}=\\mathbf{X}\\boldsymbol{\\beta}+\\mathbf{g}+\\mathbf{e},$$\n",
      "\n",
      "-   $\\mathbf{g}=\\mathbf{M}\\boldsymbol{\\alpha}$ has\n",
      "\n",
      "    -   null means and\n",
      "\n",
      "    -   covariance matrix\n",
      "\n",
      "        $$\\begin{aligned}\n",
      "        Var(\\mathbf{g}|\\mathbf{M}) \n",
      "        &=Var(\\mathbf{M}\\boldsymbol{\\alpha})\\\\\n",
      "        &=\\mathbf{M}Var(\\boldsymbol{\\alpha})\\mathbf{M}'.\n",
      "        \\end{aligned}$$\n",
      "\n",
      "\n",
      "-   Then, in both models ([eq:2-1]) and ([eq:GM]), the mean of\n",
      "    $\\mathbf{y}$ is $\\mathbf{X}\\boldsymbol{\\beta}$ and\n",
      "\n",
      "-   the covariance matrix is\n",
      "    $$Var(\\mathbf{y}|\\mathbf{M})=\\mathbf{M}Var(\\boldsymbol{\\alpha})\\mathbf{M}'+\\mathbf{I}\\sigma_{e}^{2}.$$\n",
      "    Thus, these two models are linearly equivalent and will lead to the\n",
      "    same inferences.\n",
      "\n",
      "-   When the number of markers is large relative to the size of\n",
      "    $\\mathbf{g}$, BLUP of $\\mathbf{g}$ can be obtained efficiently\n",
      "    @VanRaden:2008:J-Dairy-Sci:18946147\n",
      "    [@Stranden:2009:J-Dairy-Sci:19448030] by solving the MME that\n",
      "    correspond to ([eq:GM]).\n",
      "\n",
      "-   Under some assumptions,\n",
      "\n",
      "    $$\\sigma_{\\alpha}^{2}=\\frac{\\sigma_{g}^{2}}{\\sum_{j}2p_{j}(1-p_{j})}$$\n",
      "\n",
      "-   So,\n",
      "\n",
      "    $$\\begin{aligned}\n",
      "    Var(\\mathbf{g}|\\mathbf{M}) \n",
      "    & = \\frac{\\mathbf{MM}'}{\\sum_{j}2p_{j}(1-p_{j})}\\sigma_{g}^{2}\\nonumber \\\\\n",
      "    & = \\mathbf{G}\\sigma_{g}^{2}.\n",
      "    \\end{aligned}$$"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "BLUP combining genotype and pedigree relationships\n",
      "--------------------------------------------------\n",
      "\n",
      "-   Suppose $\\mathbf{g}$ is partitioned as\n",
      "    $$\\mathbf{g}=\\left[\\begin{array}{c}\n",
      "    \\mathbf{g}_{1}\\\\\n",
      "    \\mathbf{g}_{2}\n",
      "    \\end{array}\\right]=\\left[\\begin{array}{l}\n",
      "    \\mathbf{g}_{1}\\\\\n",
      "    \\mathbf{M}_{2}\\boldsymbol{\\alpha}\n",
      "    \\end{array}\\right],$$\n",
      "\n",
      "\n",
      "-   $\\mathbf{g_{1}}$ are the genomic BVs of the animals with missing\n",
      "    genotypes $\\mathbf{M}_{1}$\n",
      "\n",
      "-   $\\mathbf{g}_{2}$ are the BVs of those with observed genotypes\n",
      "    $\\mathbf{M}_{2}$.\n",
      "\n",
      "-   Following Legarra et al. @Legarra:2009:J-Dairy-Sci:19700729, the\n",
      "    vector $\\mathbf{g}_{1}$ can be written as\n",
      "\n",
      "    $$\\begin{aligned}\n",
      "    \\mathbf{g}_{1} \n",
      "    & = \\mathbf{A}_{12}\\mathbf{A}_{22}^{-1}\\mathbf{g}_{2}+(\\mathbf{g}_{1}-\\mathbf{A}_{12}\\mathbf{A}_{22}^{-1}\\mathbf{g}_{2})\\nonumber \\\\\n",
      "    & = \\hat{\\mathbf{g}}_{1}+\\boldsymbol{\\epsilon},\n",
      "    \\end{aligned}$$"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "-   $\\mathbf{A}_{ij}$ are partitions of $\\mathbf{A}$ that correspond to\n",
      "    $\\mathbf{g}_{1}$ and $\\mathbf{g}_{2}$.\n",
      "\n",
      "-   The first term in ([eq:g1.2]) is the best linear predictor (BLP) of\n",
      "    $\\mathbf{g}_{1}$ given $\\mathbf{g}_{2}$,\n",
      "\n",
      "-   the second is a residual.\n",
      "\n",
      "-   It is easy to see that $\\boldsymbol{\\epsilon}$ in ([eq:g1.2]) is\n",
      "    uncorrelated to $\\mathbf{g}_{2}$,\n",
      "\n",
      "-   therefore if $\\mathbf{g}_{1}$ and $\\mathbf{g}_{2}$ are multivariate\n",
      "    normal, $\\mathbf{\\epsilon}$ and $\\mathbf{g}_{2}$ are independent."
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "-   Consider first the conditional distribution of $\\mathbf{g}_{1}$\n",
      "    given $\\mathbf{P}$. Then, as expected, the\n",
      "\n",
      "    -   variance of $\\mathbf{g}_{1}$ is\n",
      "\n",
      "        $$\\begin{aligned}\n",
      "        Var(\\mathbf{g_{1}|}\\mathbf{P}) \n",
      "        & = [\\mathbf{A}_{12}\\mathbf{A}_{22}^{-1}\\mathbf{A}_{21}+(\\mathbf{A}_{11}-\\mathbf{A}_{12}\\mathbf{A}_{22}^{-1}\\mathbf{A}_{21})]\\sigma_{g}^{2}\\\\\n",
      "        & = \\mathbf{A}_{11}\\sigma_{g}^{2},\\end{aligned}$$\n",
      "\n",
      "        where the first term of ([eq:varg1]) is the variance of the\n",
      "        $\\hat{{\\bf{g}}}_{1}$ and the second term is the\n",
      "        variance of $\\boldsymbol{\\epsilon}$.\n",
      "\n",
      "    -   Similarly,\n",
      "        $Var(\\mathbf{g}_{2}|\\mathbf{P})=\\mathbf{A}_{22}\\sigma_{g}^{2}$.\n",
      "\n",
      "-   Consider now the conditional distribution of $\\mathbf{g}_{1}$ given\n",
      "    $\\mathbf{M}_{2}$.\n",
      "\n",
      "-   Note that, given the observed genotypes $\\mathbf{M}_{2}$, the\n",
      "    distribution of $\\mathbf{g}_{2}$ changes to a multivariate normal\n",
      "    with\n",
      "\n",
      "    -   mean $\\mathbf{0}$ and\n",
      "\n",
      "    -   covariance matrix\n",
      "        $\\mathbf{M}_{2}\\mathbf{M}_{2}'\\sigma_{\\alpha}^{2}$."
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "- As explained below, this change in the distribution of\n",
      "    $\\mathbf{g}_{2}$, produces an associated change in the distribution\n",
      "    of $\\mathbf{g}_{1}$ to a normal with\n",
      "\n",
      "  - mean $\\mathbf{0}$ and\n",
      "  - covariance matrix:\n",
      "$$\n",
      "Var(\\mathbf{g_{1}|M}_{2})=\\mathbf{A}_{12}\\mathbf{A}_{22}^{-1}\\mathbf{M}_{2}\\mathbf{M}'_{2}\\mathbf{A}_{22}^{-1}\\mathbf{A}_{21}\\sigma_{\\alpha}^{2}+(\\mathbf{A}_{11}-\\mathbf{A}_{12}\\mathbf{A}_{22}^{-1}\\mathbf{A}_{21})\\sigma_{g}^{2},\n",
      "$$\n",
      "  - where now the vector $\\hat{\\mbox{g}}_{1}=\\mathbf{A}_{12}\\mathbf{A}_{22}^{-1}\\mathbf{g}_{2}$ has covariance matrix given by the first term of ([eq:varg|z]),\n",
      "  - because $\\boldsymbol{\\epsilon}$ is independent of $\\mathbf{g}_{2}$, the second term of ([eq:varg|z]) remains identical to that of ([eq:varg1])."
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "-   Similarly, the covariance between $\\mathbf{g}_{1}$ and $\\mathbf{g}_{2}$ conditional on $\\mathbf{M}_{2}$ is\n",
      "$$Cov(\\mathbf{g}_{1},\\mathbf{g}_{2})=\\mathbf{A}_{12}\\mathbf{A}_{22}^{-1}\\mathbf{M}_{2}\\mathbf{M}'_{2}\\sigma_{\\alpha}^{2}.$$"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "-   Further, assuming ([eq:MarkerGenVar]), the above results can be\n",
      "    combined to show that conditional on $\\mathbf{M}_{2}$,\n",
      "\n",
      "    -   $\\mathbf{g}$ has a multivariate normal distribution with null\n",
      "        mean and covariance matrix:\n",
      "\n",
      "        $$Var(\\mathbf{g}|\\mathbf{M}_{2})=\\mathbf{H}=\\left[\\begin{array}{cc}\n",
      "        \\mathbf{A}_{12}\\mathbf{A}_{22}^{-1}\\mathbf{G}\\mathbf{A}_{22}^{-1}\\mathbf{A}_{21}+(\\mathbf{A}_{11}-\\mathbf{A}_{12}\\mathbf{A}_{22}^{-1}\\mathbf{A}_{21}) & \\mathbf{A}_{12}\\mathbf{A}_{22}^{-1}\\mathbf{G}\\\\\n",
      "        \\mathbf{G}\\mathbf{A}_{22}^{-1}\\mbox{A}_{21} & \\mathbf{G}\n",
      "        \\end{array}\\right]\\sigma_{g}^{2},$$\n",
      "\n",
      "        where\n",
      "        $\\mathbf{G=}\\mathbf{M}_{2}\\mathbf{M}_{2}'/[\\sum2p_{i}(1-p_{i})]$.\n",
      "\n",
      "    -   The inverse of this matrix is needed to setup the MME, and this\n",
      "        is computed as\n",
      "        $$\\mathbf{H}^{-1}=\\mathbf{A}^{-1}+\\left[\\begin{array}{ll}\n",
      "        \\mathbf{0} & \\mathbf{0}\\\\\n",
      "        \\mathbf{0} & \\mathbf{G}^{-1}-\\mathbf{A}_{22}^{-1}\n",
      "        \\end{array}\\right].$$"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "-   Note that this requires computing both $\\mathbf{G}^{-1}$ and\n",
      "    $\\mathbf{A}_{22}^{-1}$, which are dense and not easy to compute.\n",
      "\n",
      "-   Due to the increased adoption of SNP genotyping in livestock,\n",
      "    $\\mathbf{G}^{-1}$ and $\\mathbf{A}_{22}^{-1}$ are becoming too large\n",
      "    for SS-GBLUP to remain computationally feasible (e.g. \\>100,000\n",
      "    animals).\n",
      "\n",
      "-   A second problem in SS-GBLUP is related to the scaling that is done\n",
      "    using the SNP frequencies.\n",
      "\n",
      "-   As mentioned earlier, when all data that were used for selection are\n",
      "    available for computing the conditional mean, it can be computed as\n",
      "    if selection had not taken place @Gianola.D.ea:1986\n",
      "    [@Im.S.ea:1989; @Sorensen.DA.ea:2001].\n",
      "\n",
      "-   If selection has taken place, this requires using SNP frequencies\n",
      "    from the founders, as these frequencies are not changed by\n",
      "    selection.\n",
      "\n",
      "-   In most situations, however, SNP genotypes are not available in the\n",
      "    founders and frequencies observed in the genotyped animals are used,\n",
      "    which can lead to biased evaluations, particularly in a multi-breed\n",
      "    context.\n",
      "\n",
      "-   An approach very similar to that of using ([eq:g1.2]) to model\n",
      "    missing genotypes was proposed by Fernando (see equation (43)\n",
      "    in@gianola06:_genom) in the context of genomic prediction using\n",
      "    kernel regression, where missing genotypes were replaced by their\n",
      "    conditional expectation computed using all available information, in\n",
      "    contrast to using BLP as in SS-GBLUP.\n",
      "\n",
      "-   Also, a residual that is similar to $\\boldsymbol{\\epsilon}$ was\n",
      "    included in the model.\n",
      "\n",
      "-   When these residuals are modeled exactly, the inverse of their\n",
      "    covariance matrix is not sparse and SS-GBLUP would not be\n",
      "    computationally feasible."
     ]
    }
   ],
   "metadata": {}
  }
 ]
}